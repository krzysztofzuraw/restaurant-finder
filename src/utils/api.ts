import axios from 'axios';

import { configService } from '~src/services';

const config = configService.getConfig();

export const agentInstance = axios.create({ baseURL: config.GEOCODING_URL });
