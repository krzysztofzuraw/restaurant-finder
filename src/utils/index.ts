export * from './tests';
export * from './store';
export * from './api';
export * from './marbles';
