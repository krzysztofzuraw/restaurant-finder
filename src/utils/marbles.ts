import { ActionsObservable, Epic, StateObservable } from 'redux-observable';
import { Subject } from 'rxjs';

import { TestScheduler } from 'rxjs/testing';
import Types from 'Types';
import { rootReducer } from '~src/store';

const testScheduler = new TestScheduler((actual, expected) => expect(actual).toEqual(expected));
const initialState = rootReducer(undefined, {} as Types.RootAction);

export const marbleTest = (
  actions: string,
  expected: string,
  values: { [marble: string]: Types.RootAction },
  services: {},
  epic: Epic
) =>
  testScheduler.run(({ cold, expectObservable }) => {
    const state$ = new StateObservable(new Subject<Types.RootState>(), initialState);
    const action$ = new ActionsObservable(cold<Types.RootAction>(actions, values));
    const dependencies = {
      geocodingService: {
        searchForPlaces: jest.fn(),
      },
      loggingService: {
        logError: jest.fn(),
      },
      configService: {
        getConfig: jest.fn(),
      },
      mapsService: {
        initMap: jest.fn(),
        placeCurrentLocationMarker: jest.fn(),
      },
      geolocationService: {
        getCurrentPosition: jest.fn(),
      },
      ...services,
    };
    const output$ = epic(action$, state$, dependencies);

    expectObservable(output$).toBe(expected, values);
  });
