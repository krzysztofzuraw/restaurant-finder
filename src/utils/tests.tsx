import * as React from 'react';

import { AppTheme } from '~src/theme';

export const setupComponent = (children: JSX.Element) => <AppTheme>{children}</AppTheme>;
