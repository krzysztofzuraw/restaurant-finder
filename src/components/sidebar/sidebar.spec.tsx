import * as React from 'react';
import { cleanup, render } from 'react-testing-library';

import { setupComponent } from '~src/utils';
import { Sidebar } from '.';

afterEach(cleanup);

describe('Sidebar Component', () => {
  it('should render correctly without places', () => {
    const component = render(setupComponent(<Sidebar places={[]} />));
    expect(component).toMatchSnapshot();
  });

  // it('should render correctly with places', () => {
  //   const mockedPlaces = [
  //     {
  //       name: 'Vaffanapoli',
  //       distance: '100 meters',
  //       addressPartOne: 'Włodkowica 5/5',
  //       addressPartTwo: '50-250 Wrocław',
  //     },
  //     {
  //       name: 'Gniazdo',
  //       distance: '1km',
  //       addressPartOne: 'Opera 5/5',
  //       addressPartTwo: '50-250 Wrocław',
  //     },
  //   ];
  //   const component = render(setupComponent(<Sidebar placeResults={mockedPlaces} />));
  //   expect(component).toMatchSnapshot();
  // });
});
