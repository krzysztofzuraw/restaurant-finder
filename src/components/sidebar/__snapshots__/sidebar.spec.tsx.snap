// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar Component should render correctly with places 1`] = `
Object {
  "baseElement": <body>
    .emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background-image: url();
  background-repeat: no-repeat;
  background-color: #FFFFFF;
  background-position: 30px;
  height: 100px;
  padding-left: 90px;
  padding-top: 12px;
  padding-right: 12px;
  border-bottom: 0.75px solid #D6D6D6;
  -webkit-transition: box-shadow 0.3s;
  transition: box-shadow 0.3s;
}

.emotion-4 > div:first-child {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-letter-spacing: 0.0032em;
  -moz-letter-spacing: 0.0032em;
  -ms-letter-spacing: 0.0032em;
  letter-spacing: 0.0032em;
  font-weight: 600;
  line-height: 25px;
}

.emotion-4:hover {
  -webkit-transform: translateY(-1px);
  -ms-transform: translateY(-1px);
  transform: translateY(-1px);
  box-shadow: 0 0 11px rgba(33,33,33,0.2);
}

.emotion-0 {
  color: #333333;
  margin-right: auto;
  font-size: 1.3em;
}

.emotion-1 {
  justify-self: flex-end;
  color: #666666;
  font-size: 0.75em;
}

.emotion-3 {
  margin-top: 5px;
  color: #333333;
  font-weight: 400;
  font-size: 0.75em;
  -webkit-letter-spacing: 0.0032em;
  -moz-letter-spacing: 0.0032em;
  -ms-letter-spacing: 0.0032em;
  letter-spacing: 0.0032em;
}

.emotion-2 {
  margin-bottom: 2.5px;
}

<div>
      <ul>
        <li
          class="emotion-4"
        >
          <div>
            <h2
              class="emotion-0"
            >
              Vaffanapoli
            </h2>
            <h6
              class="emotion-1"
            >
              100 meters
            </h6>
          </div>
          <div
            class="emotion-3"
          >
            <div
              class="emotion-2"
            >
              Włodkowica 5/5
            </div>
            <div>
              50-250 Wrocław
            </div>
          </div>
        </li>
        <li
          class="emotion-4"
        >
          <div>
            <h2
              class="emotion-0"
            >
              Gniazdo
            </h2>
            <h6
              class="emotion-1"
            >
              1km
            </h6>
          </div>
          <div
            class="emotion-3"
          >
            <div
              class="emotion-2"
            >
              Opera 5/5
            </div>
            <div>
              50-250 Wrocław
            </div>
          </div>
        </li>
      </ul>
    </div>
  </body>,
  "container": <div>
    <ul>
      <li
        class="css-14f4z7v"
      >
        <div>
          <h2
            class="css-195ik64"
          >
            Vaffanapoli
          </h2>
          <h6
            class="css-x4e08b"
          >
            100 meters
          </h6>
        </div>
        <div
          class="css-cpc7te"
        >
          <div
            class="css-1flldo8"
          >
            Włodkowica 5/5
          </div>
          <div>
            50-250 Wrocław
          </div>
        </div>
      </li>
      <li
        class="css-14f4z7v"
      >
        <div>
          <h2
            class="css-195ik64"
          >
            Gniazdo
          </h2>
          <h6
            class="css-x4e08b"
          >
            1km
          </h6>
        </div>
        <div
          class="css-cpc7te"
        >
          <div
            class="css-1flldo8"
          >
            Opera 5/5
          </div>
          <div>
            50-250 Wrocław
          </div>
        </div>
      </li>
    </ul>
  </div>,
  "debug": [Function],
  "getAllByAltText": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Sidebar Component should render correctly without places 1`] = `
Object {
  "baseElement": <body>
    <div>
      <ul />
    </div>
  </body>,
  "container": <div>
    <ul />
  </div>,
  "debug": [Function],
  "getAllByAltText": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
